### PR TITLE
Alex/app_v1

### PR DESCRIPTION
**Pull request by augustin-cramer**

### CODE REVIEW

**Comment by augustin-cramer** [comment]
will need to add `nltk.download('stopwords')` and `nltk.download('punkt')` somewhere for the code to be able to run in a docker environment

File: .DS_Store
Comment (augustin-cramer) [inline .DS_Store:1]: remove

File: polarization values/Polarization between ['GUA'] VS ['DM'] ; axis = 1, companies = all, percentiles = [10, 90], with parliament = False.csv
Comment (augustin-cramer) [inline polarization values/Polarization between ['GUA'] VS ['DM'] ; axis = 1, companies = all, percentiles = [10, 90], with parliament = False.csv:1]: remove

File: polarization values/Polarization between ['Lab'] VS ['Con'] ; axis = 1, companies = all, percentiles = [10, 90], with parliament = True.csv
Comment (augustin-cramer) [inline polarization values/Polarization between ['Lab'] VS ['Con'] ; axis = 1, companies = all, percentiles = [10, 90], with parliament = True.csv:1]: remove

File: src/Polarization/cos_pol.py [hunk 1]
```diff
@@ -289,4 +281,5 @@ def change_year(old_year):
         )
 
     # Display the figure
-    fig.show()
+    return fig
+    return fig
```
Comment (augustin-cramer) [inline src/Polarization/cos_pol.py]: remove one of these

File: streamlit_app.py
Comment (augustin-cramer) [inline streamlit_app.py:1]: remove

File: word analysis values/processed yearly data ; year = 2017, model = 7, with parliament = False
Comment (augustin-cramer) [inline word analysis values/processed yearly data ; year = 2017, model = 7, with parliament = False:1]: remove

File: word analysis values/processed yearly data ; year = 2018, model = 8, with parliament = False
Comment (augustin-cramer) [inline word analysis values/processed yearly data ; year = 2018, model = 8, with parliament = False:1]: remove

File: word analysis values/var embed real ; current year = 2018, previous year = 2017
Comment (augustin-cramer) [inline word analysis values/var embed real ; current year = 2018, previous year = 2017:1]: remove

File: src/Word_analysis/words_variation.py [hunk 2]
```diff
@@ -216,29 +227,34 @@ def word_variations(
         year = year + 18090
     i = eval(str(year)[-1:])
 
-    # Assuming you have a dictionary `model_words` with keys as years and values as the corresponding model for that year
     path_1 = f"word analysis values/processed yearly data ; year = {year}, model = {i}, with parliament = {with_parliament}"
     if not os.path.exists(path_1):
+        st.write((f"processing year {year}"))
+        st.write((f"processing year {year}"))
```
Comment (augustin-cramer) [inline src/Word_analysis/words_variation.py]: remove one of these two

File: src/Word_analysis/words_variation.py [hunk 3]
```diff
@@ -216,29 +227,34 @@ def word_variations(
...
         print(f"processing year {year}")
         current_df = process_year_data(year, models_w[i], with_parliament)
         current_df.to_csv(path_1, index=False)
     else:
+        st.write(f"{year} already processed")
```
Comment (augustin-cramer) [inline src/Word_analysis/words_variation.py:330]: same

File: src/Word_analysis/words_variation.py [hunk 4]
```diff
@@ -216,29 +227,34 @@ def word_variations(
...
+        st.write(f"{year} already processed")
         print(f"{year} already processed")
         current_df = pd.read_csv(path_1)
 
     path_2 = f"word analysis values/processed yearly data ; year = {year-1}, model = {i-1}, with parliament = {with_parliament}"
     if not os.path.exists(path_2):
+        st.write(f"processing year {year-1}")
```
Comment (augustin-cramer) [inline src/Word_analysis/words_variation.py:336]: same

File: src/Word_analysis/axis_variation.py [hunk 5]
```diff
@@ -170,7 +170,8 @@ def project_variation_on_axis(
     fig.update_yaxes(showgrid=True, gridwidth=0.5, gridcolor="gray")
 
     # Display the figure
-    fig.show()
+    return fig
+    return fig
```
Comment (augustin-cramer) [inline src/Word_analysis/axis_variation.py]: remove one of these

File: src/Polarization/cos_pol.py [hunk 6]
```diff
@@ -22,7 +20,8 @@ def draw_cos_pol(
     if not axis:
         raise ValueError("It only works on an axis")
 
-    companies = ["all"]
+    companies = "all"
+    companies = "all"
```
Comment (augustin-cramer) [inline src/Polarization/cos_pol.py]: remove one of these

**Review by augustin-cramer — COMMENTED** [review]
some files to delete.
quality stuff <3

**Review by alexmaxad — COMMENTED** [review]
Changes reviewed and removed the wrongly merged files

**Comment by alexmaxad** [comment]
zut ça passe pas black. Je vais appliquer et re commit

**Comment by alexmaxad** [comment]
J'attends pour merge avec le main

File: requirements.txt
Comment (augustin-cramer) [inline requirements.txt:1]: we might have some conflicts there

File: src/Axes/models_dataframes.py [hunk 7]
```diff
@@ -1,25 +1,34 @@
...
 
 # DataFrames Preparation and Processing
 
 dfs = []
 # Loop through a specified range to load and project sentence embeddings for each year
 for i in range(14):
+    if i < 10:
+        year = 2010 + i
+    else:
+        year = 20110 + (i - 10)
```
Comment (augustin-cramer) [inline src/Axes/models_dataframes.py:27]: could be better to change the files names instead hahaha

File: src/Polarization/cos_pol.py [hunk 8]
```diff
@@ -16,20 +21,64 @@ def draw_cos_pol(
...
 
     sources = left_side + right_side
 
     if not os.path.exists(
         f"polarization values/Polarization between {left_side} VS {right_side} ; axis = {axis}, companies = {companies}, percentiles = {percentiles}, with parliament = {with_parliament}.csv"
     ):
-        print("computin polarization")
+        print("computin polarization...")
+        st.write("computin polarization...")
```
Comment (augustin-cramer) [inline src/Polarization/cos_pol.py]: remove the print and add a "g"

**Review by augustin-cramer — COMMENTED** [review]
nice

**Review by nael-farhan — COMMENTED** [review]
ca part

**Review by nael-farhan — APPROVED** [review]
avec approve

**Pull request merged**